fbb/wl2k.go

```diff
@@ -223,12 +227,13 @@ func (s *Session) Exchange(conn net.Conn) (stats TrafficStats, err error) {
 			err = io.EOF
 		}
 
-		if err != io.EOF {
+		switch err {
+		case io.EOF, io.ErrUnexpectedEOF:
```

Comment:
I take it when a connection closes cleanly, it does not cause `io.EOF`? That's not a control-flow signal like in file processing?

Reply:
That's right.

`io.EOF` is in fact a control-flow signal, but at the tcp connection level. It is returned from `conn.Read` and `conn.Write`. However a successful B2F/FBB exchange ends with either peer sending the `FQ` command, and because of this we would never see an io.EOF error unless the connection was closed prematurely.